app/controllers/topics_controller.rb

```diff
@@ -9,12 +9,19 @@ class TopicsController < ApplicationController
   # GET /topics/autocomplete_topic
   def autocomplete_topic
     term = params[:term]
-    @topics = if term && !term.empty?
-                policy_scope(Topic.where('LOWER("name") like ?', term.downcase + '%')).order('"name"')
-              else
-                []
-              end
-    render json: autocomplete_array_json(@topics)
+    if term && !term.empty?
+      @topics = policy_scope(Topic.where('LOWER("name") like ?', term.downcase + '%')).order('"name"')
+      @mapTopics = @topics.select { |t| t.metacode.name == 'Metamap' }
+      # prioritize topics which point to maps, over maps
+      @exclude = @mapTopics.length > 0 ? @mapTopics.map(&:name) : ['']
+      @maps = policy_scope(Map.where('LOWER("name") like ? AND name NOT IN (?)', term.downcase + '%', @exclude)).order('"name"')
+    else
+      @topics = []
+      @maps = []
+    end
+    @all= @topics.concat(@maps).sort { |a, b| a.name <=> b.name }
```

Comment:
One rails best practise we've been moving to at work is using maximum one instance variable per controller. There's various reasons why, most of which I don't think we care about in Metamaps because our codebase isn't crazy big

But in this action in particular, we're only using one instance variable, so we could probably drop the @ on the other ones. it's just like a form of communication; which variable is being "rendered"